### PR TITLE
Multinode ensemble

### PR DESCRIPTION
**Pull request by jmccreight**
This implements a new run mode for ensembles that says good riddance to PBS job arrays and their horrid performance on cheyenne:
The basic idea:
1: request all your resources from PBS for running the ensemble,
2: handle the ensemble run within wrfhydropy

The wrfhydropy side of the equation is agnostic about the scheduler and how one identifies the resources/nodes which are available. It simply requires a (team_dict)[https://github.com/NCAR/wrf_hydro_py/compare/master...jmccreight:multinode_dart?expand=1#diff-13b3a869d60a38ccdf356fd4fc9a65c6R71] which is documented in the code following that link. Basically, the ensemble run is divided up to different teams. The teams have separate resources/nodes and the teams work in parallel processing their assigned ensemble members sequentially. Entry and exit commands are parsed semicolon-wise and run undermpi sequentially on each teams first node as well. The exe_cmd is specified by the user to match the flavor of MPI being used.

### CODE REVIEW

File: wrfhydropy/core/ensemble.py [hunk 1]
```diff
@@ -54,10 +54,114 @@ def parallel_run(arg_dict):
     else:
         os.chdir(str(pathlib.Path(arg_dict['ens_dir']) / arg_dict['member'].run_dir))
     mem_pkl = pickle.load(open("WrfHydroSim.pkl", "rb"))
+
     mem_pkl.run()
     return mem_pkl.jobs[0].exit_status
 
 
+def parallel_teams_run(arg_dict):
```
Comment (jmills-ncar) [inline wrfhydropy/core/ensemble.py]: Why not expose some of this documentation as a docstring?

File: wrfhydropy/core/ensemble.py [hunk 2]
```diff
@@ -54,10 +54,114 @@ def parallel_run(arg_dict):
...
+    """Parallelizable function for teams to run an EnsembleSimuation."""
```
Comment (jmills-ncar) [inline wrfhydropy/core/ensemble.py]: You'll have to walk me through this, I'm kind of lost. All I really have to say about that then is we need to document it well.

**Comment by tjmills** [comment]
I guess these suggestions were a bit much :)

**Comment by jmccreight** [comment]
No, just not a priority! I do use this and need to get it in. ..

**Pull request merged**